source/spirv_target_env.cpp

```diff
@@ -220,3 +220,31 @@ bool spvIsOpenCLEnv(spv_target_env env) {
   }
   return false;
 }
+
+bool spvIsWebGPUEnv(spv_target_env env) {
+  switch (env) {
+    case SPV_ENV_UNIVERSAL_1_0:
```

Comment:
I guess it's deliberate that there is no "default" case.  That requires this code to be updated whenever a new environment is added.  Ok. I guess that makes sense.  (Otherwise I would make a catchall default clause.  But it's safer the way you have it here.)

Reply:
spv_target_env isn't an enum class so you don't get covering warnings I believe. Although, in this case the default case would just be the same as the non-default return of false so I'm not sure if it matters.

Reply:
For this I was following the pattern established by the other spvIs*Env functions in the file. I agree with the assessment that this isn't throwing any warnings due to it being a C-style enum, so there isn't an enforcement that all possible values are tested. I assume the not default, but instead a final return statement is likely due to some combination of compiler and flags not complaining about no return statement when just using default.

I think the advantage of not just iterating the positive cases, is that it makes it easier for a dev to scan for the enum they care about. A find query will hit the enum regardless of it being a positive or negative case, as opposed to having to figure out it is caught by the final return.